### PR TITLE
automate getting discourse embed url

### PR DESCRIPTION
**Pull request by sckott**
This makes the following change for the discourse JS snippet at the bottom of blog posts

- group 1. posts before discourse came along - no discourse embed code
- group 2. posts that have topicid in their yaml header - get discourse embed code with 
- group 3. posts we're trying to automate embeds for, no topicid and newer than 2019 (to differentiate from older posts that also don't have topicid in their yaml)

to do:

- [x] make change for technotes file
- [x] make change for commcalls file
- [x] remove dummy blog post

### CODE REVIEW

**Comment by maelle** [comment]
maybe there should be a comment partial and that'd be added to tech notes and comm calls too (instead of copy-pasting all lines)?

**Comment by maelle** [comment]
I made the partial, the original duplication was a bit my fault.

File: themes/ropensci/layouts/partials/comments.html [hunk 1]
```diff
@@ -0,0 +1,38 @@
+<!-- discourse comments -->
+<!-- 
+group 1. posts before discourse came along - no discourse embed code
+group 2. posts that have topicid in their yaml header - get discourse embed code with topocid
+group 3. posts we're trying to automate embeds for, no topicid and newer than 2019
+  (to differentiate from older posts that also don't have topicid in their yaml)
+-->
+
+{{ with .topicid }}
```
Comment (maelle) [inline themes/ropensci/layouts/partials/comments.html]: it's an useful pattern cf https://gohugo.io/templates/introduction/#example-1-with and https://gohugo.io/templates/introduction/#example-2-with--else

File: themes/ropensci/layouts/partials/comments.html [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        topicId: {{ . }} };
+      (function() {
+        var d = document.createElement('script'); d.type = 'text/javascript'; d.async = true;
+        d.src = DiscourseEmbed.discourseUrl + 'javascripts/embed.js';
+        (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(d);
+      })();
+      </script>
+        </div>
+        {{ else }}
+      {{ if (gt .date.Year 2019) }}
```
Comment (maelle) [inline themes/ropensci/layouts/partials/comments.html]: It's Date for the page but date in Params :grimacing: 

I use https://gohugo.io/templates/template-debugging/ to see this.

**Comment by maelle** [comment]
I have a general question about the Discourse embed, why does it always first work when the PR is merged i.e. not on the Netlify preview URL? (which is good) Is it a setting on the Discourse side?

**Comment by maelle** [comment]
An old post with no topicid is /blog/2012/11/26/is-invasive/

File: themes/ropensci/layouts/partials/comments.html (hunk 2 above)
Comment (sckott) [reply themes/ropensci/layouts/partials/comments.html]: okay, thanks

File: themes/ropensci/layouts/partials/comments.html (hunk 1 above)
Comment (sckott) [reply themes/ropensci/layouts/partials/comments.html]: nice

**Comment by sckott** [comment]
thanks @maelle 

- old posts: https://deploy-preview-633--ropensci.netlify.com/blog/2012/11/26/is-invasive/ correctly has no discourse html
- posts that have a topicid: https://deploy-preview-633--ropensci.netlify.com/blog/2020/02/05/ozunconf19/ correctly has discourse html with topicid
- new posts without topicid: https://deploy-preview-633--ropensci.netlify.com/blog/2020/03/04/testing-123/ this now has the wrong url - the html has `discourseEmbedUrl: "https://deploy-preview-633--ropensci.netlify.com"` - so just the base url for the site, not the url for the post itself

File: themes/ropensci/layouts/partials/comments.html [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      <div class="col-offset-2 col-8 top-4">
+        <div id='discourse-comments'></div>
+        <script type="text/javascript">
+        DiscourseEmbed = { discourseUrl: 'https://discuss.ropensci.org/',
+          discourseEmbedUrl: {{ .Page.Permalink | absURL }} };
```
Comment (maelle) [inline themes/ropensci/layouts/partials/comments.html]: Maybe remove absURL?
In any case you can print the context .Page here (debugging link from an earlier comment) to see whaf variable to use?

Comment (sckott) [reply themes/ropensci/layouts/partials/comments.html]: it was working before i'm pretty sure. that line of code `discourseEmbedUrl: {{ .Page.Permalink | absURL }} }` is the same, but maybe something else changes

Comment (maelle) [reply themes/ropensci/layouts/partials/comments.html]: right! I see what happened, I'll fix it. The permalink variable isn't available from the partial so I'll need to pass it. Sorry!

**Comment by maelle** [comment]
@sckott it should be fixed now. :smile_cat:

**Comment by sckott** [comment]
awesome, thanks @maelle

**Pull request merged**